src/promptflow/promptflow/azure/operations/_run_operations.py

```diff
@@ -995,3 +1000,43 @@ def _resolve_flow_definition_resource_id(self, run: Run):
         workspace_id = self._workspace._workspace_id
         location = self._workspace.location
         return f"azureml://locations/{location}/workspaces/{workspace_id}/flows/{run._flow_name}"
+
+    def download(
+        self, run: Union[str, Run], output: Optional[Union[str, Path]] = None, overwrite: Optional[bool] = False
+    ) -> str:
+        """Download the data of a run, including input, output, snapshot and other run information.
+
+        :param run: The run name or run object
+        :type run: Union[str, ~promptflow.entities.Run]
+        :param output: The output directory. Default to be current working directory if not specified.
+        :type output: Optional[str]
+        :param overwrite: Whether to overwrite the existing run folder. Default to be False.
+        :type overwrite: Optional[bool]
+        :return: The run directory path
+        :rtype: str
+        """
+        import platform
+
+        run = Run._validate_and_return_run_name(run)
+        output_directory = Path(".") if output is None else Path(output)
```

Comment:
shall the default output path be  ~/.promptflow/runs?

Reply:
that would be equal to automatically running pf run create, which seems a bit weird? Although convenient.

Reply:
actually i feel like it's not bad, updated.